lib/http/connection.rb

```diff
@@ -132,6 +132,10 @@ def close
       @pending_request  = false
     end
 
+    def finished?
```

Comment:
I think `finished?` is a little ambiguous, since it could mean the connection itself rather than a request. How about...
```suggestion
    def finished_request?
```